selfdrive/car/volkswagen/carstate.py

```diff
@@ -114,26 +115,20 @@ def update(self, pt_cp, cam_cp, ext_cp, trans_type):
       if ret.cruiseState.speed > 90:
         ret.cruiseState.speed = 0
 
-    # Update control button states for turn signals and ACC controls.
-    self.buttonStates["accelCruise"] = bool(pt_cp.vl["GRA_ACC_01"]["GRA_Tip_Hoch"])
-    self.buttonStates["decelCruise"] = bool(pt_cp.vl["GRA_ACC_01"]["GRA_Tip_Runter"])
-    self.buttonStates["cancel"] = bool(pt_cp.vl["GRA_ACC_01"]["GRA_Abbrechen"])
-    self.buttonStates["setCruise"] = bool(pt_cp.vl["GRA_ACC_01"]["GRA_Tip_Setzen"])
-    self.buttonStates["resumeCruise"] = bool(pt_cp.vl["GRA_ACC_01"]["GRA_Tip_Wiederaufnahme"])
-    self.buttonStates["gapAdjustCruise"] = bool(pt_cp.vl["GRA_ACC_01"]["GRA_Verstellung_Zeitluecke"])
+    # Update button states for turn signals and ACC controls, capture all ACC button state/config for passthrough
     ret.leftBlinker = bool(pt_cp.vl["Blinkmodi_02"]["Comfort_Signal_Left"])
     ret.rightBlinker = bool(pt_cp.vl["Blinkmodi_02"]["Comfort_Signal_Right"])
-
-    # Read ACC hardware button type configuration info that has to pass thru
-    # to the radar. Ends up being different for steering wheel buttons vs
-    # third stalk type controls.
-    self.graHauptschalter = pt_cp.vl["GRA_ACC_01"]["GRA_Hauptschalter"]
-    self.graTypHauptschalter = pt_cp.vl["GRA_ACC_01"]["GRA_Typ_Hauptschalter"]
-    self.graButtonTypeInfo = pt_cp.vl["GRA_ACC_01"]["GRA_ButtonTypeInfo"]
-    self.graTipStufe2 = pt_cp.vl["GRA_ACC_01"]["GRA_Tip_Stufe_2"]
-    # Pick up the GRA_ACC_01 CAN message counter so we can sync to it for
-    # later cruise-control button spamming.
-    self.graMsgBusCounter = pt_cp.vl["GRA_ACC_01"]["COUNTER"]
+    self.gra_stock_values = pt_cp.vl["GRA_ACC_01"]
+    buttonEvents = []
+    for button in MQB_BUTTONS:
```

Comment:
Nice clean up! Can we use `create_button_event` here?

Reply:
To summarize our DMs, I'll keep this in draft until you settle on the best way to merge the Tesla/global button state implementations, and then I'll sync up with that.

Reply:
Let's merge this with Tesla's implementation as VW uses different bits for its buttons too. We can clean up and unify for all cars in another PR. Is this ready for review?

Reply:
It's complete, but I haven't driven it yet. Give me the weekend to make sure cancel spam and resume spam are solid.